Core/Src/monitor.c

```diff
@@ -139,6 +147,71 @@ void vPedalsMonitor(void* pv_params)
 		// MAX_ADC_VAL_12B) && 	!is_timer_active(&diff_timer)) 	start_timer(&diff_timer,
 		// PEDAL_FAULT_TIME); else 	cancel_timer(&diff_timer);
 
+
+		/* Evaluate accelerator faults */
+		if (is_timer_expired(&oc_timer))
+			//todo queue fault
```

Comment:
Can we move all of the fault evaluation for pedals to a helper function? This is a fairly large piece of code that can be easily moved to a `static void eval_pedal_fault()` function that might help with readability of the task

Reply:
Also seems like we use the same logic for brakes and pedals, we could use the same function for both and just pass in the different values (and maybe thresholds)

Reply:
added a helper function. But it has extra parameters - void eval_pedal_fault(Sensor_1, Sensor_2, diff_timer, sc_timer, oc_timer). I was wondering if the interface can be updated accordingly